Code/Framework/AzCore/AzCore/Component/ComponentApplication.cpp

```diff
@@ -646,6 +651,106 @@ namespace AZ
+                // and store it in the path variable
+                startupCfgPath = Settings::UnquoteArgument(argument.m_value);
+            }
+
+            return true;
+        };
+
+        // Parses the command line arguments without using any heap allocations
+        // This attempts to read the startup config file from the command line
+        Settings::ParseCommandLine(argc, argv, commandLineParserSettings);
+
+        // Try to the open up the Startup config file
+        constexpr auto openMode = AZ::IO::OpenMode::ModeRead;
+        AZ::IO::SystemFileStream startupCfgStream(startupCfgPath.c_str(), openMode);
+        if (!startupCfgStream.IsOpen())
+        {
+            return;
+        }
+
+        auto& allocatorManager = AZ::AllocatorManager::Instance();
+        // Now parse the config file settings without using heap allocations as well
+        Settings::ConfigParserSettings configParserSettings;
+        configParserSettings.m_parseConfigEntryFunc =
+            [&allocatorManager](const Settings::ConfigParserSettings::ConfigEntry& configEntry) -> bool
+        {
+            // If a key in the config file is of the form with "allocator_tracking_<allocator_name>,
+            // try to convert the value into a boolean. If the value is true
+            // An entry of the allocator name is registered with the AllocatorManager
+            // which is used to turn on Allocator Tracking when an allocator with the specified
+            // name is registered
+            constexpr AZStd::string_view AllocatorTrackingPrefix = "allocator_tracking_";
+            if (AZ::StringFunc::StartsWith(configEntry.m_keyValuePair.m_key, AllocatorTrackingPrefix))
+            {
+                AZStd::string_view allocatorName = configEntry.m_keyValuePair.m_key.substr(AllocatorTrackingPrefix.size());
+                if (!allocatorName.empty())
+                {
+                    // Attempt to convert the string directly to an AllocationRecords::Mode enum
+                    // This occurs in the case when the string matches the name of an enum option
+                    AZ::Debug::AllocationRecords::Mode mode = allocatorManager.GetDefaultTrackingMode();
+                    if (!AZ::ConsoleTypeHelpers::ToValue(mode, configEntry.m_keyValuePair.m_value))
+                    {
+                        // If the conversion from string to AllocationsRecords::Mode failed, then
+                        // try to parse the value as a bool and if the bool is true, set the AllocationRecords::Mode
+                        // to record full records
+                        if (bool boolMode{}; AZ::ConsoleTypeHelpers::ToValue(boolMode, configEntry.m_keyValuePair.m_value))
+                        {
+                            mode = boolMode ? AZ::Debug::AllocationRecords::Mode::RECORD_FULL
+                                            : AZ::Debug::AllocationRecords::Mode::RECORD_NO_RECORDS;
+                        }
```

Comment:
I am going to update this wiki page with samples of the value that the "allocator_tracking_<allocator-name>" configuration can accept.
https://github.com/o3de/o3de/wiki/Allocator-Tagging-Guide#checking-the-allocator-memory-at-runtime

Reply:
It would also be nice to have a link from the docs ( https://docs.o3de.org/docs/user-guide/programming/memory/ ) to the wiki page, because I never would've thought to look to the wiki for that type of documentation. I literally just went looking for memory debugging information yesterday and that doc page is the only place I found. :)

Reply:
Yeah, I also need to update some of that documentation as well